creator/creator.go

```diff
@@ -346,9 +346,16 @@ func (c *Creator) Context() DrawContext {
 	return c.context
 }
 
-// Call before writing out. Takes care of adding headers and footers, as well
-// as generating front Page and table of contents.
-func (c *Creator) finalize() error {
+// Finalize renders all blocks to the creator pages. In addition, it takes care
+// of adding headers and footers, as well as generating the front page,
+// table of contents and outlines.
+// Finalize is automatically called before writing the document out. Calling the
+// method manually can be useful when adding external pages to the creator,
+// using the AddPage method, as it renders all creator blocks to the added
+// pages, without having to write the document out.
+// NOTE: TOC and outlines are generated only if the AddTOC and AddOutlines
```

Comment:
If there is no content in the TOC/outlines, it does not get added either, might be worth mentioning that?

Reply:
Sure. I adapted the note to also include the ways to add content to TOCs/outlines.